lib/include/jxl/decode.h

```diff
@@ -913,6 +953,144 @@ JXL_EXPORT JxlDecoderStatus JxlDecoderSetJPEGBuffer(JxlDecoder* dec,
  */
 JXL_EXPORT size_t JxlDecoderReleaseJPEGBuffer(JxlDecoder* dec);
 
+/**
+ * Sets output buffer for box output codestream.
+ *
+ * The data is owned by the caller and may be used by the decoder until
+ * JxlDecoderReleaseBoxBuffer is called or the decoder is destroyed or reset so
+ * must be kept alive until then.
+ *
+ * @param dec decoder object
+ * @param data pointer to next bytes to write to
+ * @param size amount of bytes available starting from data
+ * @return JXL_DEC_ERROR if output buffer was already set and
+ * JxlDecoderReleaseBoxBuffer was not called on it, JXL_DEC_SUCCESS otherwise
+ */
+JXL_EXPORT JxlDecoderStatus JxlDecoderSetBoxBuffer(JxlDecoder* dec,
+                                                   uint8_t* data, size_t size);
+
+/**
+ * Releases buffer which was provided with JxlDecoderSetBoxBuffer.
+ *
+ * Calling JxlDecoderReleaseBoxBuffer is required whenever
+ * a buffer is already set and a new buffer needs to be added with
+ * JxlDecoderSetBoxBuffer, but is not required before JxlDecoderDestroy or
+ * JxlDecoderReset.
+ *
+ * Calling JxlDecoderReleaseBoxBuffer when no buffer is set is
+ * not an error and returns 0.
+ *
+ * @param dec decoder object
+ * @return the amount of bytes the decoder has not yet written to of the data
+ * set by JxlDecoderSetBoxBuffer, or 0 if no buffer is set or
+ * JxlDecoderReleaseBoxBuffer was already called.
+ */
+JXL_EXPORT size_t JxlDecoderReleaseBoxBuffer(JxlDecoder* dec);
+
+/**
+ * Configures whether to get boxes in raw mode or in uncompressed
+ * mode.
+ *
+ * The default mode is non-raw (JXL_FALSE), where "brob" boxes are transparently
```

Comment:
Would the default still be non-raw if brotli-decoder support is not enabled/compiled? I would propose that the default is true so you have the same behavior whether you have support enabled or not. Applications that don't support decoding "brob" boxes would declare so by calling this function; applications that do support calling brotli (or just iterate over all the boxes without looking into their contents) would not call it.

I understand that the common case of a decoder getting an exif box would need to call this function to enable brotli-decoding; but that would give the application a point of failure where to detect that it is not supported.

Reply:
I don't mind changing the default mode to raw, and don't do brotli compression in the API implementation by default.

@veluca93 @jonsneyers any objections to this from your side?

Reply:
Ok for a low-level api. In a future high-level api (that is only available in non-size-constrained full builds), we can have convenience functions for applications that only need simple functionality (i.e. getting a decoded image and uncompressed metadata from a file on disk).

Reply:
I think it's OK.

Reply:
Done and renamed to JxlDecoderDecompressBoxes